```{r global_options, include=FALSE}
library(magrittr)
library(tidyr)
library(dplyr)
library(ggplot2)
library(cowplot)
library(reshape)
```
##Plotting codons used in each strain

Here we go reading up all my tables and merging 'em together.  I haven't incorporated read() or the join function yet, but its faster cause smaller files:
```{r}
PB1 <- merge_all(list(read.csv(file='H2N2PB1.tidy.csv', head=TRUE, sep='\t'), 
                      read.csv(file='H1N1PB1.tidy.csv', head=TRUE, sep='\t'), 
                      read.csv(file='H3N2PB1.tidy.csv', head=TRUE, sep='\t')), 
                 all.x=TRUE, all.y=TRUE)
codonvalues <- read.csv(file='codonvalues.csv', head=TRUE, sep='\t')
```

Now I gotta table with: strain, sero, year, coryear, codon, numcodon, numaa.  I also have another table holding a bunch of values associated with each codon.

Now I can plot!

__Number of each codon found in strain__
```{r fig.height=14, fig.width=14}
ggplot(PB1, aes(x=coryear, y=(numcodon), color=sero)) + 
  geom_point() + 
  geom_smooth(method='lm', se=FALSE) +
  background_grid(major = "xy", minor = "none") + 
  facet_wrap(~codon)
```


__Proportion of each codon used for its amino acid in strain__
```{r fig.height=14, fig.width=14}
ggplot(PB1, aes(x=coryear, y=(numcodon/numaa), color=sero)) + 
  geom_point() + 
  geom_smooth(method='lm', se=FALSE) +
  background_grid(major = "xy", minor = "none") + 
  facet_wrap(~codon)
```

__Pulling out linear regression__
```{r}
PB1 %>% group_by(sero, codon) %>% do(lm.result = lm((numcodon) ~ coryear, data=.)) -> fitted.models
intslope <- do(fitted.models, data.frame(sero=.$sero,codon=.$cod, 
                             intercept=coef(.$lm.result)[1], 
                             slope=coef(.$lm.result)[2]))
#there aren't that many coefficients I think? It looks as if there's only two. Where are things going wrong...
rm(fitted.models)
intslope
```

Nice!  slopes and intercetps GOT!
Next up: p values

*Note by Claus:* You should be able to obtain p values directly from the linear regression using `p.value = summary(.$lm.result)$coef[4,2]` in the above code.


```{r}
PB1 %>% group_by(sero, codon) %>%
  do(cor.res = cor.test(.$numcodon, .$coryear)) -> cors
corp <- do(cors, data.frame(sero=.$sero, codon=.$codon, cor=.$cor.res$estimate, p=.$cor.res$p.value))
rm(cors)
corp
```

I mostly just copy-pasted the stats stuff.  I think I understand them, but I'll have to mess around with them a bit more. 

Anyways, now ima dump that into a table with CAI and rtAI.  I'm also including a table that shows the change in codon usage between my PB1 construct and the WT Udorn PB1.  rtAI is calculated using the newer, larger sequencing library.

``` {r}
codonstats <- select(corp, sero, codon, p) %>% 
  left_join(select(intslope, codon, sero, slope)) %>% 
  left_join(select(codonvalues, aa, codon, AIFNratiolib2, AnoIFNratiolib2, AIFNreadslib2, AnoIFNreadslib2, AIFNratiolib1, AnoIFNratiolib1, AIFNreadslib1, AnoIFNreadslib1, CAI)) %>%
  left_join(read.csv(file='construct_v_WT_change.csv', head=TRUE, sep='\t'))
```

Here are plots comparing slope, rtAI, and CAI.  I'm expecting positive slopes for rtAI vs slope, negative for the rest I guess.

```{r fig.width=10}
ggplot(codonstats, aes(x=AIFNratiolib2/AnoIFNratiolib2, y=slope, label=codon, color=aa)) + 
  geom_point() + 
  background_grid(major = "xy", minor = "none") + 
  geom_text(size = 5) + 
  facet_wrap(~sero)

ggplot(codonstats, aes(x=CAI, y=slope, label=codon, color=aa)) + 
  geom_point() + 
  background_grid(major = "xy", minor = "none") + 
  geom_text(size = 5) + 
  facet_wrap(~sero)

codonstats %>% filter(sero == "H3N2") %>% 
  ggplot(aes(x=AIFNratiolib1/AnoIFNratiolib1, y=CAI, label=codon, color=aa)) + 
  geom_point() + 
  background_grid(major = "xy", minor = "none") + 
  geom_text(size = 5) + 
  facet_wrap(~sero)

```

Here are plots comparing the change between my older CAI construct and the WT with the rtAI, CAI and slope.

```{r fig.width=10}
codonstats %>% filter(sero == "H3N2") %>% 
  ggplot(aes(x=change_con, y=AIFNratiolib2/AnoIFNratiolib2, label=codon, color=aa)) + 
  geom_point() + 
  background_grid(major = "xy", minor = "none") + 
  geom_text(size = 5)

codonstats %>% filter(sero == "H3N2") %>% 
  ggplot(aes(x=change_con, y=CAI, label=codon, color=aa)) + 
  geom_point() + 
  background_grid(major = "xy", minor = "none") + 
  geom_text(size = 5)

codonstats %>% filter(sero == "H3N2") %>% 
  ggplot(aes(x=change_con, y=slope, label=codon, color=aa)) + 
  geom_point() + 
  background_grid(major = "xy", minor = "none") + 
  geom_text(size = 5)
```

Here are plots comparing the change between my construct exactly replicating newer H3N2 codon usage and the WT with the rtAI, CAI and slope.

```{r fig.width=10}
codonstats %>% filter(sero == "H3N2") %>% 
  ggplot(aes(x=change_new, y=AIFNratiolib2/AnoIFNratiolib2, label=codon, color=aa)) + 
  geom_point() + 
  background_grid(major = "xy", minor = "none") + 
  geom_text(size = 5)

codonstats %>% filter(sero == "H3N2") %>% 
  ggplot(aes(x=change_new, y=CAI, label=codon, color=aa)) + 
  geom_point() + 
  background_grid(major = "xy", minor = "none") + 
  geom_text(size = 5)

codonstats %>% filter(sero == "H3N2") %>% 
  ggplot(aes(x=change_new, y=slope, label=codon, color=aa)) + 
  geom_point() + 
  background_grid(major = "xy", minor = "none") + 
  geom_text(size = 5)
```

Comparing the change in codon usage from WT for both constructs.

```{r fig.width=10}
codonstats %>% filter(sero == "H3N2") %>% 
  ggplot(aes(x=change_new, y=change_con, label=codon, color=aa)) + 
  geom_point() + 
  background_grid(major = "xy", minor = "none") + 
  geom_text(size = 5)
```

I'm really not sure how to interpret this, though...