```{r global_options, include=FALSE}
library(magrittr)
library(tidyr)
library(ggplot2)
library(cowplot)
library(reshape)
library(MASS)
library(plyr)
library(dplyr)
library(scales)
library(grid)
```

```{r}
PB1 <- read.csv(file='../sequences/H3N2PB1.fasta.individualcodoncounts', head=TRUE, sep='\t')
```


_CAI test_
```{r}
codonvalues <- 
  read.csv(file='../tRNA-seq results/codonvalues.csv', head=TRUE, sep=',') %>%
  select(c(codon, aa, CAI, CAImax))


PB1 %>% 
  filter(codon!='ATG', codon!='TGG', codon!='TAA', codon!='TAG', codon!='TGA', year<=29) -> 
  filterPB1

codonvaluesbysite=NULL
pos=0
for(i in 1:764){
  pos = pos + 1
  codonvaluesbysite <- rbind(codonvaluesbysite, mutate(codonvalues, position=pos))
}
rm(pos, i)


statsR = NULL #store correlation coefficients
statsslope = NULL #store slopes

for(i in 1:1000){
  rando <- codonvaluesbysite %>% group_by(aa, position) %>% mutate(randomscore=as.numeric(sample(CAI)), randomhigh=max(CAI))
  filterPB1 %>% left_join(rando) %>% group_by(year, strain) %>% 
    summarize(score=(prod(randomscore)**(1/length(position)))/(prod(randomhigh)**(1/length(position)))) -> rtAIscores

  R <- cor(rtAIscores$year, rtAIscores$score) #calculate correlation coefficient
  statsR <- rbind(statsR, data.frame(R)) #store correlation coefficient
  
  slope <- coef(lm(score ~ year, rtAIscores))['year'] #calculate slope
  statsslope <- rbind(statsslope, data.frame(slope)) #store slope
  
}

pop_sd <- sd(statsR$R)*sqrt((length(statsR$R)-1)/(length(statsR$R)))
pop_mean <- mean(statsR$R)
sum(abs(statsR$R)>=0.914)/nrow(statsR)
statsR %>% ggplot(aes(x=R)) + geom_density() + geom_vline(xintercept = 0.914, linetype=2)
```


_rtAI test_
```{r}
codonvalues <- 
  read.csv(file='../tRNA-seq results/codonvalues.csv', head=TRUE, sep=',') %>%
  mutate(logratio = log(AIFNratiosum/AnoIFNratiosum)) %>%
  select(c(codon, aa, logratio))

PB1 %>% 
  filter(year<=29) -> 
  filterPB1

codonvaluesbysite=NULL
pos=0
for(i in 1:764){
  pos = pos + 1
  codonvaluesbysite <- rbind(codonvaluesbysite, mutate(codonvalues, position=pos))
}
rm(pos, i)

statsR = NULL #store correlation coefficients
statsslope = NULL #store slopes

for(i in 1:1000){
  rando <- codonvaluesbysite %>% group_by(aa, position) %>% mutate(randomscore=sample(logratio))
  filterPB1 %>% left_join(rando) %>% group_by(year, strain) %>% 
    summarize(score=mean(randomscore)) -> rtAIscores
  
  R <- cor(rtAIscores$year, rtAIscores$score) #calculate correlation coefficient
  statsR <- rbind(statsR, data.frame(R)) #store correlation coefficient
  
  slope <- coef(lm(score ~ year, rtAIscores))['year'] #calculate slope
  statsslope <- rbind(statsslope, data.frame(slope)) #store slope
  
}

pop_sd <- sd(statsR$R)*sqrt((length(statsR$R)-1)/(length(statsR$R)))
pop_mean <- mean(statsR$R)
sum(abs(statsR$R)>=0.902)/nrow(statsR)
statsR %>% ggplot(aes(x=R)) + geom_density() + geom_vline(xintercept = 0.902, linetype=2)
```



_GC test_
```{r}
codonvalues <- 
  read.csv(file='../tRNA-seq results/codonvalues.csv', head=TRUE, sep=',') %>%
  select(c(codon, aa, GC))

PB1 %>% 
  filter(year<=29) -> 
  filterPB1
rm(PB1)

codonvaluesbysite=NULL
pos=0
for(i in 1:764){
  pos = pos + 1
  codonvaluesbysite <- rbind(codonvaluesbysite, mutate(codonvalues, position=pos))
}
rm(pos, i)

statsR = NULL #store correlation coefficients
statsslope = NULL #store slopes

for(i in 1:1000){
  rando <- codonvaluesbysite %>% group_by(aa, position) %>% mutate(randomscore=sample(GC))
  filterPB1 %>% left_join(rando) %>% group_by(year, strain) %>% 
    summarize(score=mean(randomscore)) -> rtAIscores
  
  R <- cor(rtAIscores$year, rtAIscores$score) #calculate correlation coefficient
  statsR <- rbind(statsR, data.frame(R)) #store correlation coefficient
  
  slope <- coef(lm(score ~ year, rtAIscores))['year'] #calculate slope
  statsslope <- rbind(statsslope, data.frame(slope)) #store slope
  
}

pop_sd <- sd(statsR$R)*sqrt((length(statsR$R)-1)/(length(statsR$R)))
pop_mean <- mean(statsR$R)
sum(abs(statsR$R)>=0.5403361)/nrow(statsR)
statsR %>% ggplot(aes(x=R)) + geom_density() + geom_vline(xintercept = 0.5403361, linetype=2)
```
