```{r global_options, include=FALSE}
library(magrittr)
library(tidyr)
library(dplyr)
library(ggplot2)
library(cowplot)
library(reshape)
library(MASS)
library(scales)
library(grid)
```
```{r}
PB1<- merge_all(list(
  (read.csv(file='../sequences/H1N1PB1.fasta.codoncounts', head=TRUE, sep='\t') %>% 
     mutate(sero='H1N1') %>% 
     filter(year>=1930) %>% 
     mutate(coryear=ifelse(year<=1960, year-1918, ifelse(year>=1960, year-1945, 0)))), 
  (read.csv(file='../sequences/H2N2PB1.fasta.codoncounts', head=TRUE, sep='\t') %>% 
     mutate(sero='H2N2') %>% 
     mutate(coryear=year-1957)),
  (read.csv(file='../sequences/H3N2PB1.fasta.codoncounts', head=TRUE, sep='\t') %>%
     mutate(sero='H3N2') %>% 
     mutate(coryear=year-1968))), all.x=TRUE, all.y=TRUE)

PB1 <- PB1 %>% left_join(read.csv(file='../tRNA-seq results/codonvalues.csv', head=TRUE, sep=','))

constructs <- read.csv(file='../sequences/constructs.fasta.codoncounts', head=TRUE, sep='\t') %>% 
  mutate(codon=tolower(codon)) %>% 
  left_join(read.csv(file='../tRNA-seq results/codonvalues.csv', head=TRUE, sep=','))
```

*CAI for all serotypes*
```{r}
PB1 %>% 
  filter(codon != 'ATG', codon!='TGG', codon!='TAA', codon!='TAG', codon!='TGA') %>%
  group_by(sero, year, strain) %>% 
  mutate(cods = sum(numcodon), initC = CAI ** numcodon, initCm = CAImax ** numcodon) %>% 
  summarize(score=((prod(initC))**mean(1/cods))/((prod(initCm))**mean(1/cods))) -> CAIscores

CAIscores %>% 
  ggplot(aes(x=year, y=score, color=sero)) + 
  geom_point(size=.5) +
  ylab('CAI') +
  xlab('Isolation Date (year)') +
  ylim(.565, .605) +
  theme(axis.text=element_text(size=12), 
        axis.title=element_text(size=14,face="bold"), 
        plot.title=element_text(size=10, face="bold"),
        legend.text=element_text(size=12),
        legend.title=element_text(size=12,face="bold"))
```