```{r global_options, include=FALSE}
library(magrittr)
library(tidyr)
library(dplyr)
library(ggplot2)
library(cowplot)
library(reshape)
library(MASS)
library(scales)
library(grid)
```
```{r}
PB1<- merge_all(list(
  (read.csv(file='../sequences/H1N1PB1.fasta.codoncounts', head=TRUE, sep='\t')%>%mutate(sero='H1N1')), 
  (read.csv(file='../sequences/H2N2PB1.fasta.codoncounts', head=TRUE, sep='\t')%>%mutate(sero='H2N2')),
  (read.csv(file='../sequences/H3N2PB1.fasta.codoncounts', head=TRUE, sep='\t')%>%mutate(sero='H3N2'))), all.x=TRUE, all.y=TRUE)

PB1 <- PB1 %>% left_join(read.csv(file='../tRNA-seq results/codonvalues.csv', head=TRUE, sep=','))

constructs <- read.csv(file='../sequences/constructs.fasta.codoncounts', head=TRUE, sep='\t') %>% 
  mutate(codon=tolower(codon)) %>% 
  left_join(read.csv(file='../tRNA-seq results/codonvalues.csv', head=TRUE, sep=','))
```

*CAI for H3N2*
```{r}
PB1 %>% 
  filter(codon != 'ATG', codon!='TGG', codon!='TAA', codon!='TAG', codon!='TGA', sero=='H3N2') %>%
  group_by(sero, year, strain) %>% 
  mutate(cods = sum(numcodon), initC = CAI ** numcodon, initCm = CAImax ** numcodon) %>% 
  summarize(score=((prod(initC))**mean(1/cods))/((prod(initCm))**mean(1/cods))) -> CAIscores

PB1 %>% 
  filter(codon != 'ATG', codon!='TGG', codon!='TAA', codon!='TAG', codon!='TGA', year<=1998, sero=='H3N2') %>%
  group_by(sero, year, strain) %>% 
  mutate(cods = sum(numcodon), initC = CAI ** numcodon, initCm = CAImax ** numcodon) %>% 
  summarize(score=((prod(initC))**mean(1/cods))/((prod(initCm))**mean(1/cods))) -> CAI30yr

constructs %>%
  filter(codon != 'ATG', codon!='TGG', codon!='TAA', codon!='TAG', codon!='TGA') %>%
  group_by(sero, year, strain) %>% 
  mutate(cods = sum(numcodon), initC = CAI ** numcodon, initCm = CAImax ** numcodon) %>% 
  summarize(score=((prod(initC))**mean(1/cods))/((prod(initCm))**mean(1/cods))) -> conCAIscores

siglabels <- function(){
    function(x) format(x,nsmall = 3,scientific = FALSE)
}

CAIscores %>% 
  ggplot(aes(x=year-1968, y=score)) + 
  geom_hline(data=conCAIscores, aes(yintercept=score), linetype=3, color="#000000") +
  geom_smooth(data=CAI30yr, aes(x=year-1968, y=score), method='lm', se=FALSE, fullrange=TRUE, color='darkgrey', size=0.5) +
  geom_point(color="#3333FF", size=.5) +
  geom_point(data=conCAIscores, aes(x=year+49, y=score), size=4, shape=18, color="#000000") +
  geom_label(data=conCAIscores, aes(x=year+49, y=score, label = strain), size=4, hjust = 0, nudge_x = .05, color="#000000") +
  scale_y_continuous(labels = siglabels(), limits = c(.565, .595), name='CAI') +
  scale_x_continuous(name='Circulation Time (years)', limits = c(0, 50)) +
  theme(axis.text=element_text(size=12), 
        axis.title=element_text(size=14,face="bold"), 
        plot.title=element_text(size=10, face="bold"),
        legend.text=element_text(size=12),
        legend.title=element_text(size=12,face="bold")) + 
  theme(legend.position="none") + 
  theme(plot.margin = unit(c(0,2,0,0), "cm")) -> noclip

gt <- ggplot_gtable(ggplot_build(noclip))
gt$layout$clip[gt$layout$name=="panel"] <- "off"

ggdraw(gt)+draw_label(expression('R=-0.914 p=0.017'), x=.4, y=.45, vjust = 1, hjust = 1, size = 14, fontface = 'bold')


(cor(CAI30yr$year, CAI30yr$score))
```

*GC for H3N2*
```{r}
PB1 %>% 
  filter(codon != 'ATG', codon!='TGG', codon!='TAA', codon!='TAG', codon!='TGA', sero=='H3N2') %>%
  group_by(sero, year, strain) %>% 
  summarize(score=(sum(numcodon*GC))/sum(numcodon)) -> GCscores

PB1 %>% 
  filter(codon != 'ATG', codon!='TGG', codon!='TAA', codon!='TAG', codon!='TGA', year<=1998, sero=='H3N2') %>%
  group_by(sero, year, strain) %>% 
  summarize(score=(sum(numcodon*GC))/sum(numcodon)) -> GC30yr

constructs %>% 
  filter(codon != 'ATG', codon!='TGG', codon!='TAA', codon!='TAG', codon!='TGA', strain!='Construct #1') %>%
  group_by(sero, year, strain) %>% 
  summarize(score=(sum(numcodon*GC))/sum(numcodon)) -> conGC1scores

constructs %>% 
  filter(codon != 'ATG', codon!='TGG', codon!='TAA', codon!='TAG', codon!='TGA', strain=='Construct #1') %>%
  group_by(sero, year, strain) %>% 
  summarize(score=(sum(numcodon*GC))/sum(numcodon)) -> conGC2scores

siglabels <- function(){
    function(x) format(x,nsmall = 3,scientific = FALSE)
}

GCscores %>%
  ggplot(aes(x=year-1968, y=score)) + 
  geom_hline(data=conGC1scores, aes(yintercept=score), linetype=3, color="#000000") +
  geom_smooth(data=GC30yr, aes(x=year-1968, y=score), method='lm', se=FALSE, fullrange=TRUE, color='darkgrey', size=0.5) +
  geom_point(color = "#3333FF", size=0.5) +
  geom_point(data=conGC1scores, aes(x=year+49, y=score), size=4, shape=18, color="#000000") +
  geom_label(data=conGC2scores, aes(x=year+49, y=score, label = strain), size=4, hjust = 0, vjust = -.55, nudge_x = 0.05, color="#000000") +
  geom_label(data=conGC1scores, aes(x=year+49, y=score, label = strain), size=4, hjust = 0, nudge_x = 0.05, color="#000000") +
  scale_y_continuous(labels = siglabels(), limits = c(.415, .43), name='GC content') +
  scale_x_continuous(name='Circulation Time (years)', limits = c(0, 50)) +
  theme(axis.text=element_text(size=12), 
        axis.title=element_text(size=14,face="bold"), 
        plot.title=element_text(size=10, face="bold"),
        legend.text=element_text(size=12),
        legend.title=element_text(size=12,face="bold")) + 
  theme(legend.position="none") + 
  theme(plot.margin = unit(c(0,2,0,0), "cm")) -> noclip

gt <- ggplot_gtable(ggplot_build(noclip))
gt$layout$clip[gt$layout$name=="panel"] <- "off"


ggdraw(gt)+draw_label(expression('R=-0.562 p=0.509'), x=.4, y=.3, vjust = 1, hjust = 1, size = 14, fontface = 'bold')

(cor(GC30yr$year, GC30yr$score))
```

*rtAI for H3N2*
```{r}
PB1 %>%
  filter(sero=='H3N2') %>%
  group_by(sero, year, strain) %>%
  summarize(score=(sum(numcodon*log(AIFNratiosum/AnoIFNratiosum)))/sum(numcodon)) -> rtAIscores

constructs %>%
  group_by(sero, year, strain) %>%
  summarize(score=(sum(numcodon*log(AIFNratiosum/AnoIFNratiosum)))/sum(numcodon)) -> conrtAIscores

PB1 %>% 
  filter(year<=1997, sero=='H3N2') %>%
  group_by(sero, year, strain) %>% 
  mutate(cods = sum(numcodon), initC = CAI ** numcodon, initCm = CAImax ** numcodon) %>% 
  summarize(score=(sum(numcodon*log(AIFNratiosum/AnoIFNratiosum)))/sum(numcodon)) -> rtAI30yr
  
rtAIscores %>% ggplot(aes(x=year-1968, y=score)) + 
  geom_hline(data=conrtAIscores, aes(yintercept=score), linetype=3, color="#000000") +
  geom_smooth(data=rtAI30yr, aes(x=year-1968, y=score), method='lm', se=FALSE, fullrange=TRUE, color='darkgrey', size=0.5) +
  geom_point(color = "#3333FF", size=0.5) + 
  geom_point(data=conrtAIscores, aes(x=year+49, y=score), size=4, shape=18, color="#000000") +
  geom_label(data=conrtAIscores, aes(x=year+49, y=score, label = strain), size=4, hjust = 0, nudge_x = .05, color="#000000") +
  xlab('Circulation Time (years)') + 
  ylab('rtAI (IFN-treated/untreated)') +
  theme(axis.text=element_text(size=12), 
        axis.title=element_text(size=14,face="bold"), 
        plot.title=element_text(size=14, face="bold"),
        legend.text=element_text(size=12),
        legend.title=element_text(size=12,face="bold"))+
  theme(legend.position="none") + 
  theme(plot.margin = unit(c(0,2.2,0,0), "cm")) -> noclip

gt <- ggplot_gtable(ggplot_build(noclip))
gt$layout$clip[gt$layout$name=="panel"] <- "off"

ggdraw(gt)+draw_label(expression('R=0.902 p=0.041'), x=.6, y=.3, vjust = 1, hjust = 1, size = 14, fontface = 'bold')

cor(rtAI30yr$year, rtAI30yr$score)
```